gptme/tools/__init__.py

```diff
@@ -120,7 +122,17 @@ def execute_msg(msg: Message, confirm: ConfirmFunc) -> Generator[Message, None,
 
     for tooluse in ToolUse.iter_from_content(msg.content):
         if tooluse.is_runnable:
-            yield from tooluse.execute(confirm)
+            try:
+                for tool_response in tooluse.execute(confirm):
+                    yield tool_response.replace(call_id=tooluse.call_id)
+            except KeyboardInterrupt:
+                clear_interruptible()
+                yield Message(
+                    "system",
+                    "User hit Ctrl-c to interrupt the process",
```

Comment:
I kinda want to make these shorter

```suggestion
                    "Interrupted by user",
```

Reply:
Sort of what I did in #384

Reply:
I changed the message (and used a constant to have the same value everywhere it's needed.)